### ISSUE TITLE
About the TubeTK Hull Mask Files

### ISSUE THREAD

**Issue by dengchihwei** [batmanlab/vessel-flow-consistency-ssl#4]
Thanks for sharing the wonderful work!

However, since I see that you used a private mask files for TubeTK dataset. It will be really helpful if you could share these hull masks. 

Thanks a lot.

**Comment by rohitrango** [comment]
you can use `generate_binaryimage_tubetk.py` to generate binary masks. Let me know if this works.

**Comment by dengchihwei** [comment]
Thanks for replying.
Sorry if I miss understand. But the `generate_binaryimage_tubetk.py` file only generates the ground truth label files from the .tre file. However, I noticed that you use a new `hull` mask file for the whole MRA image at line 124 of `get_all_metrics_tubetk.py` file. I am wondering how these files are generated.

**Comment by rohitrango** [comment]
I think the`hull` mask files are simply generated by thresholding the MRA image to discard background, and slightly eroding the mask (to not consider the boundary of the brain as a vessel).

**Comment by dengchihwei** [comment]
If thresholding is applied on both original image and the vesselness response image, then the calculated metrics will also depend on the quality of the hull image, won't it? Will be appreciated if you can share a simple script to generate these hull images. Thanks.

**Comment by rohitrango** [comment]
The threshold to get the hull of the image is only applied on the original image. All methods can use this hull to discount the edge of the brain (w/ the background) being detected as vessel / ridge

**Comment by dengchihwei** [comment]
So, the hull only discounts the edge of brain or also eliminates the background? 
Since I noticed that many voxels also have very high intensities at the `skull` part. If a single threshold is used, then some of the background that `inside` brain region should be removed also. Am i understanding correctly?

**Comment by dengchihwei** [comment]
I used a brain skull-stripped mask based on T1 image as the hull mask. And then, I registered this image to the MRA image space. The mask seems pretty valid to me, as shown below.

<img width="610" alt="image" src="https://user-images.githubusercontent.com/49810478/200442140-52bbb0f9-da60-4d51-baaa-874c476d29ad.png">

However, the Dice score based on these masks only have '~0.40' dice score for trained model. So, I think the thresholded 'hull' mask that you generated might be much more precise than this image, or the hull image you generated should remove many background voxels 'inside' the brain itself? So the model can achieve reported '0.59' dice scores. So, it will be very useful if the author can share a pre-trained model or the hull mask files.

Another thing is that, I saw during the TubeTk dataset training, the training set have many overlapped images with the testing set. Since you use the first 80 images in the dataset for training.  So, I am wondering if this would cause a data leak problem?
